src/ol/tilegrid/tilegrid.js

```diff
@@ -511,24 +521,24 @@ ol.tilegrid.getForProjection = function(projection) {
 ol.tilegrid.createForExtent =
     function(extent, opt_maxZoom, opt_tileSize, opt_corner) {
   var tileSize = goog.isDef(opt_tileSize) ?
-      opt_tileSize : ol.DEFAULT_TILE_SIZE;
+      ol.size.toSize(opt_tileSize) : ol.tilegrid.DEFAULT_TILE_SIZE;
```

Comment:
It might be good to `ol.tilegrid.DEFAULT_TILE_SIZE.slice()` here in case some future code thinks it is safe to `ol.size.buffer` with the same as an `opt_size`.  I don't think the extra allocation is a concern for this function.